### PR TITLE
[pickers] Call `onChange` when selecting a picker with `changeImportance="set"`

### LINKED ISSUE #9931: [pickers] `onChange` not run when `changeImportance: 'set'`
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/laughing-lichterman-gmvscf?file=/Demo.tsx

Steps:
1. Set the `changeImportance` flag in the `shortcut` props to 'set'
2. `console.log()` the `onChange` and `onAccept` functions
3. Select a `shortcut` item in the `dateRangePicker`
4. Expect to see a log message in the console for the `onChange` function
5. Select the `OK` button
6. Expect to see a log message in the console for `onAccept` function
7. observe both `onChange` and `onAccept` are not run 

image

image

### Current behavior 😯

When setting the `changeImportance` property of the `shortcuts` object to `'set'` in the `DateRangeSelector` the `onChange` function is not run when a shortcut is selected and the `onAccept` function is not run after a shortcut is selected followed by selecting the `OK` button.

### Expected behavior 🤔

- `onChange` is run when a shortcut is selected and `changeImportance: 'set'`
- `onAccept` is run after a shortcut is selected followed by `OK` and `changeImportance: `set` 
 

### Context 🔦

Implementation requirements were:

- Update app state with the `DateRangePicker` `value` if and only if the `OK` button is selected.
- Don't close the component when a shortcut is selected only when `Cancel`, `OK`, or `clickAway` are triggered.

We were unable to satisfy the above requirements.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
Browser tested: Chrome

  System:
    OS: macOS 13.4.1
  Binaries:
    Node: 18.16.0 - ~/.asdf/installs/nodejs/18.16.0/bin/node
    Yarn: 3.5.1 - ~/.asdf/installs/yarn/1.22.19/bin/yarn
    npm: 9.5.1 - ~/.asdf/installs/nodejs/18.16.0/bin/npm
  Browsers:
    Chrome: 115.0.5790.170
    Edge: Not Found
    Safari: 16.5.2
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1 
    @emotion/styled: ^11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.8 
    @mui/core-downloads-tracker:  5.14.3 
    @mui/icons-material: ^5.14.1 => 5.14.1 
    @mui/lab: ^5.0.0-alpha.137 => 5.0.0-alpha.137 
    @mui/material: ^5.14.2 => 5.14.3 
    @mui/private-theming:  5.13.7 
    @mui/styled-engine:  5.13.2 
    @mui/system:  5.14.1 
    @mui/types:  7.2.4 
    @mui/utils:  5.13.7 
    @mui/x-data-grid:  6.10.2 
    @mui/x-data-grid-premium: ^6.10.2 => 6.10.2 
    @mui/x-data-grid-pro:  6.10.2 
    @mui/x-date-pickers:  6.10.2 
    @mui/x-date-pickers-pro: ^6.10.2 => 6.10.2 
    @mui/x-license-pro: ^6.10.2 => 6.10.2 
    @types/react: ^18.2.15 => 18.2.15 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^5.1.6 => 5.1.6 
```
</details>

### Order ID or Support key 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by flaviendelangle**
Fixes #9931

It was just a dumb mistake on my side :/

### CODE REVIEW

**Review by LukasTy — APPROVED** [review]
Nice and quick fix. 👍 
Could be worth adding tests to assert the different behaviors of `set` and `accept`. 🤔

**Comment by flaviendelangle** [comment]
I added some tests :+1:

File: packages/x-date-pickers/src/tests/describeValue/testShortcuts.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                },
+              ],
+            },
+          }}
+        />,
+      );
+
+      const shortcut = document.querySelector(
+        `.${pickersLayoutClasses.shortcuts} .${chipClasses.root}`,
+      )!;
```
Comment (LukasTy) [inline packages/x-date-pickers/src/tests/describeValue/testShortcuts.tsx:54]: Nitpick, consider replacing with :
```suggestion
      const shortcut = screen.getByRole('button', { name: 'Test shortcut' });
```

Comment (flaviendelangle) [reply packages/x-date-pickers/src/tests/describeValue/testShortcuts.tsx]: That's a lot better thanks!

**Pull request merged**